### PR TITLE
execution/stagedsync: index block-end system call logs in the parallel executor

### PR DESCRIPTION
**Pull request by mriccobene**
Previously, the parallel executor collected logs emitted by block-end `engine.Finalize` system calls but did not index them. Its per-transaction publish loop only indexed receipt logs, and block-end work has no receipt. This produced different `LogAddrIdx` and `LogTopicIdx` contents from the serial executor.

The parallel executor now indexes these logs directly on the execution loop, using the block's final virtual transaction number and skipping a second receipt cache write. Finalize messages no longer carry unused log, trace, or blob gas data; the unused log copy and `isFinalize` flag are also removed.

This surfaced in the Gnosis QA exec-from-zero [run](https://github.com/erigontech/erigon/actions/runs/33932349204): generated `idx/v3.0-logaddrs.*.ef` and `idx/v3.0-logtopics.*.ef` files differed from the published files, while trace indexes over the same step ranges matched. Gnosis calls its block reward contract at the end of nearly every block, and that call emits a log. The published files were built before parallel execution became the default.

JSON-RPC responses remain unchanged. `eth_getLogs`, `erigon_getLogs`, and Otterscan search skip the final virtual transaction. This change fixes the missing block-end entries in the log index files.

Regression tests check both address and topic indexes for serial and parallel execution. The test with repeated system calls counts index additions before storage deduplication, so duplicate processing remains detectable. The cleanup of finalize data is a pure refactor, so it uses existing executor tests instead of a separate TDD cycle.

Validation completed:

- `go test ./execution/stagedsync -short -run '^Test(Parallel|Serial|HandleMessage)' -count=1 -timeout=5m`
- `make lint`
- `make erigon integration`
- Confirmed that the parallel regression fails without the fix. Temporary fault injections also made the topic-index assertions and the test for duplicate processing fail as expected.

The full Gnosis snapshot comparison was not rerun locally.

### CODE REVIEW

**Comment by AskAlexSharov** [comment]
Fix is right: `finalVersion.TxNum` is the same block-end txNum the serial path indexes at, it runs on the exec loop that owns `sd.mem`, and `skipReceiptCache=true` avoids a second receipt-cache put at that txNum.

Three cuts:

- `exec3_parallel.go:3460` + `:3513` — after this change nothing reads `lastResult.Logs`. Its only consumer is `logs:` in the finalize `sendResult`, and the apply loop takes just `blockNum/blockGasUsed/writes/txNum` off a `*txResult` (`case *txResult:`, line 561). Drop the append and the `logs`/`traceFroms`/`traceTos`/`cumulativeBlobGasUsed` fields from that literal — that dead pair is what made the missing indexing invisible in the first place. `require.Len(t, txResult.Logs, 1)` in the new test goes with it; `indexedTxNums` is the assertion that matters.
- `exec3_parallel.go:3467` — `be.blobGasUsed` is ignored on this call: with `receipt == nil` and `skip == true`, `applyLogsAndTraces4` reads `cummulativeBlobGas` only inside `if receipt != nil`. Pass `0`, so it does not read as blob gas being accounted a second time.
- `exec3_finalize_logindex_test.go` — 131 lines for two tests whose helpers already live next door (`logEmittingSyscallEngine`, `seedLogEmittingContract`, `newResumeTestDB` in `exec3_parallel_test.go`; `newSerialFinalizeTestExec` in `exec3_serial_test.go`). Move the two tests into those files and drop the new one.

On the history-mode review comment: not this PR. The per-tx call at line 3333 has no `!HistoryExecution` guard either, so the new call matches the surrounding parallel code. If that guard is needed, it is missing for every tx and belongs in its own issue.

File: execution/stagedsync/exec3_finalize_logindex_test.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+			version:  state.Version{BlockNum: 1, TxIndex: 0, Incarnation: 1, TxNum: txTask.TxNum},
+		},
+		ExecutionResult: evmtypes.ExecutionResult{ReceiptGasUsed: 21000},
+	}
+
+	res, err := be.nextResult(t.Context(), pe, txResult, rwTx)
+	require.NoError(t, err)
+	require.NotNil(t, res)
+	require.NoError(t, res.Err)
+	require.Len(t, txResult.Logs, 1)
```
[the remark this answers is not in the document]
Comment (yperbasis) [reply execution/stagedsync/exec3_finalize_logindex_test.go]: Addressed in [f95d459c6a](https://github.com/erigontech/erigon/commit/f95d459c6acfd7723a30e9e069dc73413cce613e). The fixture now emits `LOG1` with topic `0x42`, and both serial and parallel tests assert `LogTopicIdx`. The tests pass, and both new assertions fail when topic indexing is removed.

**Review by yperbasis — APPROVED** [review]

File: execution/stagedsync/exec3_finalize_logindex_test.go [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+				db:          db,
+				engine: &logEmittingSyscallEngine{
+					Engine:   ethash.NewFaker(),
+					contract: accounts.InternAddress(logIndexContract),
+					calls:    1,
+				},
+				vmConfig: &vm.Config{},
+			},
+			doms:   domains,
+			rs:     state.NewStateV3Buffered(state.NewStateV3(domains, false, logger)),
```
[the remark this answers is not in the document]
Comment (yperbasis) [reply execution/stagedsync/exec3_finalize_logindex_test.go:100]: Added this coverage in stacked draft #24098 ([356082dc97](https://github.com/erigontech/erigon/commit/356082dc9792e264ba6f56077fdb7b84cbf77da5)). `TestParallelBlockEndLogsPreserveReceiptHistory` enables both receipt-cache persistence and the RCache history domain, publishes a regular receipt, then processes the final virtual transaction and block-end system call. It checks that the previous receipt remains unchanged in history, the latest cache value is cleared, and both log indexes contain the block-end entries. The affected package suites, lint, and both builds pass. Fault injections that skip the final receipt update or corrupt the saved history make the new test fail.

For clarity, I also tested #24067 with the skip flag removed: its current memory batch already preserves history on repeated writes at the same txNum, so removing that flag alone did not corrupt receipt history. #24098 replaces the flag with `ApplyLogIndexes` and now covers both that helper and the parallel finalization path.
